### PR TITLE
feat: add functions exercise

### PR DESCRIPTION
**Pull request by vaeng**
That has slipped, sorry.
I finished the concept but never made the respective exercise.

### CODE REVIEW

File: exercises/concept/lasagna-master/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Instructions
+
+In this exercise you are going to write some more code related to preparing and cooking your brilliant lasagna from your favorite cookbook.
+
+You have five tasks.
+The first one is related to the cooking itself, the other three are about the perfect preparation.
```
Comment (ErikSchierboom) [inline exercises/concept/lasagna-master/.docs/instructions.md:6]:
```suggestion
The first one is related to the cooking itself, the other four are about the perfect preparation.
```

File: exercises/concept/lasagna-master/.docs/introduction.md
Comment (ErikSchierboom) [inline exercises/concept/lasagna-master/.docs/introduction.md:1]: My main comment would be that this exercise is quite "meaty," in introducing a lot. Function/overloading and default parameters go well together I think. Const could potentially be its own exercise, but it's also fine to just keep it as is for now.

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/concept/lasagna-master/.docs/introduction.md
Comment (vaeng) [reply exercises/concept/lasagna-master/.docs/introduction.md:1]: I once had an extra exercise for const planned and I have evaluated C#'s "Authentication System" exercise. I am not sure it is a good fit.
Siebenschläfer also suggested, that the const could be moved, but we need to visit constexpr and const in classes again, so there will be some repetition and I planned to sprinkle it here and there to make it easier to digest.

If we keep it this way, we should watch the numbers and split some of it away, if users don't find it as well fitting as I did while writing it.

Comment (ErikSchierboom) [reply exercises/concept/lasagna-master/.docs/introduction.md:1]: Sure

**Pull request merged**